features/frameworks/TARGET_PSA/pal/pal_mbed_os_intf.cpp

```diff
@@ -135,9 +135,11 @@ int test_start(test_entry_f test_f, compliance_test_type type)
 {
     test_g = test_f;
     type_g = type;
+#if defined(MBED_CONF_RTOS_PRESENT)
     MBED_ASSERT((type > COMPLIANCE_TEST_START) && (type < COMPLIANCE_TEST_END));
     Thread thread(osPriorityNormal, TEST_STACK_SIZE, NULL);
     thread.start(main_wrapper);
     thread.join();
+#endif
```

Comment:
Do we need these ifdefs? Will this function run when there is no RTOS? What do we want this function to do if there is no RTOS?

If this function will run, and we want it to be more than a no-op, perhaps we should call `main_wrapper()` here.

Reply:
@Patater When we do "mbed test -m NUCLEO_F411RE -t GCC_ARM --app-config TESTS/configs/baremetal.json" then build tools build mbed-os and green tea test with bare-metal configuration and No RTOS.

pal_mbed_os_intf.cpp uses RTOS API which leads to undefined threads APIs and builds fails.so I added the ifdefs to make the build pass.

This is the first level changes to run green tea test for bare metal.

Reply:
Why is `pal_mbed_os_intf.cpp` included in the build at all? Is it needed for anything in baremetal?

Reply:
I am happy with this approach (no-op in `test_start()`) so long as greentea sees the PSA tests are skipped, not that they passed (because pass is meaningless if the tests did nothing).

Reply:
You could have returned a non-zero value but it is not checked anywhere...